inst/include/Rcpp/exceptions.h

```diff
@@ -89,6 +89,48 @@ namespace Rcpp {
             file_io_error("file already exists", file) {}	// #nocov end
     };
 
+    #define RCPP_ADVANCED_EXCEPTION_CLASS(__CLASS__, __WHAT__)                                                                                                                                                     \
```

Comment:
We have some years-old tests for macros for C++11 which can probably do with brush-up.  Now that we will have a decent compiler "most of the time" we may as well use it.

But we can of course not loose sight of the poor souls clinging to their old RHEL/CentOS/... systems from the last ice age.  It's is probably worth our while to ensure those build too.